File: 3737_count-subarrays-with-majority-element-i/csharp/solution.cs
public class Solution
{
    public int CountMajoritySubarrays(int[] nums, int target)
    {
        var score = nums.Select(n => n == target ? 1 : -1).ToArray();

        var prefixSums = new int[nums.Length + 1];

        for (int i = 1; i <= nums.Length; i++)
        {
            prefixSums[i] = prefixSums[i - 1] + score[i - 1];
        }

        var count = 0;
        for (int l = 0; l < prefixSums.Length; l++)
        {
            for (int r = l + 1; r < prefixSums.Length; r++)
            {
                if (prefixSums[r] > prefixSums[l])
                    count++;
            }
        }

        return count;
    }
}

File: 3737_count-subarrays-with-majority-element-i/rust/solution.rs
#[cfg(feature = "harness")]
pub struct Solution;

impl Solution {
    pub fn count_majority_subarrays(nums: Vec<i32>, target: i32) -> i32 {
        let score: Vec<i32> = nums
            .iter()
            .map(|&n| if n == target { 1 } else { -1 })
            .collect();

        let mut prefix_sums = vec![0; score.len() + 1];

        for i in 1..=nums.len() {
            prefix_sums[i] = prefix_sums[i - 1] + score[i - 1];
        }

        let mut count = 0;

        for l in 0..prefix_sums.len() {
            for r in l + 1..prefix_sums.len() {
                if prefix_sums[r] > prefix_sums[l] {
                    count += 1;
                }
            }
        }

        count
    }
}

